source/common/config/xds_source_id.cc

```diff
@@ -0,0 +1,23 @@
+#include "source/common/config/xds_source_id.h"
+
+#include "absl/strings/str_cat.h"
+
+namespace Envoy {
+namespace Config {
+
+XdsConfigSourceId::XdsConfigSourceId(absl::string_view authority_id,
+                                     absl::string_view resource_type_url)
+    : authority_id_(authority_id), resource_type_url_(resource_type_url) {}
+
+std::string XdsConfigSourceId::toKey() const {
+  // The delimiter between parts of the key.
+  static constexpr char DELIMITER[] = "+";
+
+  // Base64 URL-encoding the authority_id and resource_type_url so that the resulting string does
+  // not contain a "+" character, so that we can use it as a delimiter.
+  return absl::StrCat(Base64Url::encode(authority_id_.data(), authority_id_.size()), DELIMITER,
```

Comment:
OOC if this is no longer human-readable, why not hash the concatenated string (w/ "+" or "," as delimiter) or hash each string and combine the hash values?

Reply:
That's possible too, though I'm not sure what it adds?  i.e. what are the pros and cons of doing that over what's in this function?  I figured it doesn't matter as long as: (1) there's a function that can produce the same key for the same authority id and type and (2) the delimiter doesn't use characters in the hash, so as to not create a possibly rare but confusing situation where the authority id and type get mixed up if the DELIMITER char is present in the authority id.